app/src/main/java/com/duckduckgo/app/browser/RealTabSwipingAvailability.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser
+
+import com.duckduckgo.app.cta.db.DismissedCtaDao
+import com.duckduckgo.app.cta.model.CtaId
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+@ContributesBinding(AppScope::class)
+class RealTabSwipingAvailability @Inject constructor(
+    swipingTabsFeature: SwipingTabsFeature,
+    private val dismissedCtaDao: DismissedCtaDao,
+) : TabSwipingAvailability {
+    private val isOnboardingComplete: Boolean
+        get() = dismissedCtaDao.exists(CtaId.DAX_END)
```

Comment:
@0nko for checking if onboarding is finished, you need to check `UserStageStore.daxOnboardingActive()` instead `dismissedCtaDao.exists(CtaId.DAX_END)`. There are some scenarios where DAX_END dialog has been shown but onboarding is still active.
Also, this method is not used so not sure if it should be part of the PR?

Reply:
Thanks!